### PR TITLE
AP_Logger: simplify the message writers

### PR DESCRIPTION
**Pull request by WickedShell**
Specifically:
  - Remove some unreachable nullptr checks
  - Remove a noop stage
  - Remove unneeded initializers

### CODE REVIEW

File: libraries/AP_Logger/LoggerMessageWriter.cpp [hunk 1]
```diff
@@ -37,10 +37,6 @@ void LoggerMessageWriter_DFLogStart::process()
 {
     switch(stage) {
     case ls_blockwriter_stage_init:
-        stage = ls_blockwriter_stage_formats;
-        FALLTHROUGH;
-
-    case ls_blockwriter_stage_formats:
```
Comment (peterbarker) [inline libraries/AP_Logger/LoggerMessageWriter.cpp]: init is the stage which could go away here (if any!), not _formats.  It's just a name thing...

File: libraries/AP_Logger/LoggerMessageWriter.cpp [hunk 2]
```diff
@@ -201,14 +197,6 @@ void LoggerMessageWriter_WriteAllRallyPoints::process()
...
 
     case ar_blockwriter_stage_init:
-        if (_rally == nullptr) {
-            stage = ar_blockwriter_stage_done;
-            break;
-        }
-        stage = ar_blockwriter_stage_write_new_rally_message;
-        FALLTHROUGH;
-
-    case ar_blockwriter_stage_write_new_rally_message:
```
Comment (peterbarker) [inline libraries/AP_Logger/LoggerMessageWriter.cpp]: Again, init should go away, not new_rally_message

File: libraries/AP_Logger/LoggerMessageWriter.cpp [hunk 3]
```diff
@@ -255,15 +243,6 @@ void LoggerMessageWriter_WriteEntireMission::process() {
...
     case em_blockwriter_stage_init:
-        if (_mission == nullptr) {
-            stage = em_blockwriter_stage_done;
-            break;
-        } else {
-            stage = em_blockwriter_stage_write_new_mission_message;
-        }
-        FALLTHROUGH;
-
-    case em_blockwriter_stage_write_new_mission_message:
```
Comment (peterbarker) [inline libraries/AP_Logger/LoggerMessageWriter.cpp]: ... and again :-)

File: libraries/AP_Logger/LoggerMessageWriter.h [hunk 4]
```diff
@@ -30,12 +30,12 @@ class LoggerMessageWriter_WriteSysInfo : public LoggerMessageWriter {
 
 private:
     enum write_sysinfo_blockwriter_stage {
```
Comment (peterbarker) [inline libraries/AP_Logger/LoggerMessageWriter.h]: This should be `: uint8_t` if you feel like making the change

**Comment by WickedShell** [comment]
@peterbarker I believe I've addressed your concerns.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**